mypy/checker.py

```diff
@@ -5914,6 +5914,8 @@ def is_type_call(expr: CallExpr) -> bool:
 
         def combine_maps(list_maps: list[TypeMap]) -> TypeMap:
             """Combine all typemaps in list_maps into one typemap"""
+            if all(m is None for m in list_maps):
+                return None
             result_map = {}
             for d in list_maps:
                 if d is not None:
```

Comment:
```suggestion
            result_map = None
            for d in list_maps:
                if d is not None:
                    if result_map is None:
                        result_map = {}
```

Reply:
I'd keep my version if you don't mind: this part isn't hot (only reachable when we actually have some `type(x) <op> Something` checks), and I find it more explicit/slightly easier to follow. This shouldn't have any measurable performance impact.